### ISSUE TITLE
Pagination stops working in Channel Page when used initialMessageId

### ISSUE THREAD

**Issue by bhanu-under25** [GetStream/stream-chat-flutter#1085]
### Which packages are you using?

stream_chat_flutter, stream_chat_flutter_core

### On what platforms did you experience the issue?

iOS, Android

### What version are you using?

  - stream_chat_flutter: ^3.5.1

### What happened?

We were trying to build our own way of opening the channel page to unread message using ChannelPage with initialMessage

Ex:
```
@override
  Widget build(BuildContext context) {
    return StreamChannel(
      channel: channel,
      showLoading: false,
      initialMessageId: (channel?.state?.unreadCount ?? 0) > 0 || fromSearch
          ? initialMessage?.id
          : null,
      child: ChannelPage(
        initialMessage: initialMessage,
        prevUnreadCount: channel?.state?.unreadCount ?? 0,
      ),
    );
  }
}
```

but after navigating the pagination doesn't load other messages apart from couple of messages.

### Steps to reproduce

```bash
1. Open ChannelPage with Initial Message ID
2. Try to load above and below message by scrolling to make fetch message by pagination.
3. No other messages load and it needs to be open again for it to work.
```

### Supporting info to reproduce

- This is how the ChannelPage looks without the initialMessageId

<img width="376" alt="Screenshot 2022-04-14 at 8 11 15 PM" src="https://user-images.githubusercontent.com/92363650/163416076-48318d92-8e51-4118-b630-2fe9cd4830bf.png">

- ChannelPage after navigating by passing initial message id

https://user-images.githubusercontent.com/92363650/163416253-d3e271e2-2036-43c2-b7ae-8246acde8192.mov

As you can see only few messages gets loaded and pagination doesn't work and no others messages get fetched.

### Relevant log output

```shell
No error gets thrown in the console.
```

### Flutter doctor output

```shell
[✓] Flutter (Channel stable, 2.10.3, on macOS 12.3 21E230 darwin-arm, locale en-IN)
    • Flutter version 2.10.3 at /Users/bhanu/Flutter/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (6 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /Users/bhanu/Library/Android/sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.66.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.38.1

[✓] Connected device (2 available)
    • ASUS I005DA (mobile) • 192.168.1.147:5555 • android-arm64  • Android 11 (API 30)
    • Chrome (web)         • chrome             • web-javascript • Google Chrome 100.0.4896.88

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by imtoori** [comment]
I'll check this and let you know in the next days. Sorry about the delay

**Comment by imtoori** [comment]
this should be fixed by #1144
you'll find the fix in the next release
